packages/dso-toolkit/src/components/legend-item/legend-item.models.ts

```diff
@@ -0,0 +1,14 @@
+import { Selectable } from "../selectable/selectable.models";
+
+export interface LegendItem<TemplateFnReturnType> {
+  body?: TemplateFnReturnType;
+  disabled?: boolean;
+  disabledMessage?: string;
+  dsoMouseEnter?: () => void;
+  dsoMouseLeave?: () => void;
+  dsoRemoveClick?: (e: CustomEvent<MouseEvent>) => void;
```

Comment:
Wij geven nooit direct het originele event mee als custom event details maar altijd een object:

```
export interface LegendItem<TemplateFnReturnType> {
  dsoRemoveClick?: (e: CustomEvent<LegendItemRemoveClickEvent>) => void;
}

export interface LegendItemRemoveClickEvent {
  originalEvent: MouseEvent;
}
```

Op deze manier zijn we toekomstvast. Eventuele uitbreidingen betekenen geen breaking change, immers kunnen we een bestaand model uitbreiden.

Reply:
Prima, maar ik kom nog wel wat uitzonderingen tegen ..., waaronder

<img width="791" alt="Screenshot 2024-04-11 at 08 54 13" src="https://github.com/dso-toolkit/dso-toolkit/assets/4646174/78161c92-429d-4255-8d36-d754663d1833">

Reply:
✅

Reply:
Ik weet het 😞 

![image](https://github.com/dso-toolkit/dso-toolkit/assets/9696261/58af5dc8-8a7e-4e22-8a5f-b4d39a3db6e3)

Iets met schade en schande... ⚫ 🪶...